pkg/kubelet/container/helpers_test.go

```diff
@@ -1129,3 +1130,399 @@ func TestHasAnyActiveRegularContainerStarted(t *testing.T) {
 		})
 	}
 }
+
+func TestExpandContainerCommandOnlyStatic(t *testing.T) {
+	testCases := []struct {
+		name     string
+		command  []string
+		envs     []v1.EnvVar
+		expected []string
+	}{
+		{
+			name:     "no command",
+			command:  nil,
+			envs:     nil,
+			expected: nil,
+		},
+		{
+			name:    "static env expansion",
+			command: []string{"echo", "$(FOO)", "$(BAR)"},
+			envs: []v1.EnvVar{
+				{Name: "FOO", Value: "foo-value"},
+				{Name: "BAR", Value: "bar-value"},
+			},
+			expected: []string{"echo", "foo-value", "bar-value"},
+		},
+		{
+			name:    "missing env variable",
+			command: []string{"echo", "$(FOO)", "$(MISSING)"},
+			envs: []v1.EnvVar{
+				{Name: "FOO", Value: "foo-value"},
+			},
+			expected: []string{"echo", "foo-value", "$(MISSING)"}, // should this behave as "echo", "foo-value", "" ?
```

Comment:
It is better to move the comment you added here to the docs of `ExpandContainerCommandOnlyStatic`. Something like "missing env vars will stay not expanded". I don't think the behavior can be changed now